File: Napkin/wwwroot/typescript/model/item.ts
class Item {
	id: string;
	raw: Point[];
	shape: Shape;
	text: string;
	fontSizeK: number;
	lineArrowEnd: boolean;
	lineArrowStart: boolean;

	constructor(){
		this.id = this.generateNewId();
		this.raw = [];
		this.shape = Shape.Original;
		this.text = '';
		this.fontSizeK = 1;
		this.lineArrowEnd = false;
		this.lineArrowStart = false;
	}

	private generateNewId() : string {
		return 'id' + Math.round(Math.random() * 1000000);
	}

	record(point: Point) {
		this.raw.push(new Point(point.x, point.y)); // Kinda sketchy but need to make sure we dont retain old refs
	}
	
	isLine(): boolean {
		return ;
	}
};

File: Model/Item.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Newtonsoft.Json;

namespace wsweb.Model
{
    public class Item
    {
		[JsonProperty("id")]
		public string Id { get; set; }

		[JsonProperty("raw")]
		public List<Point> Raw { get; set; }

		[JsonProperty("shape")]
		public Shape Shape { get; set; }

		[JsonProperty("text")]
		public string Text { get; set; }

		[JsonProperty("fontSizeK")]
		public decimal FontSizeK { get; set; }

		[JsonProperty("lineArrowEnd")]
		public bool LineArrowEnd { get; set; }

		[JsonProperty("lineArrowStart")]
		public bool LineArrowStart { get; set; }
    }
}
